Tests/ClientRuntimeTests/NetworkingTests/Http/NIO/NIOHTTPClientStreamBridgeTests.swift

```diff
@@ -0,0 +1,142 @@
+        let maxDataSize = 65_536  // 64 kb
+
+        // Create & fill a buffer with random bytes, for use in later test setup
+        // Random buffer is reused because creating random data is slow
+        // We are responsible for deallocating it
+        let randomBuffer = UnsafeMutablePointer<UInt8>.allocate(capacity: maxDataSize)
+        defer { randomBuffer.deallocate() }
+
+        for i in 0..<maxDataSize {
+            randomBuffer[i] = UInt8.random(in: UInt8.min...UInt8.max)
+        }
+
+        // Run this test repeatedly to help uncover any problems
+        let numberOfRuns = 100
+
+        for run in 1...numberOfRuns {
+            // Test with varying data sizes up to the maximum
+            let dataSize = min(run * 1000, maxDataSize)
+
+            // Create original test data
+            let originalData = Data(bytes: randomBuffer, count: dataSize)
+
+            // Create a mock AsyncHTTPClient response with the test data
+            let allocator = ByteBufferAllocator()
+            var buffer = allocator.buffer(capacity: originalData.count)
+            buffer.writeBytes(originalData)
+
+            let response = AsyncHTTPClient.HTTPClientResponse(
+                version: .http1_1,
+                status: .ok,
+                headers: [:],
+                body: .bytes(buffer)
+            )
+
+            let resultStream = await NIOHTTPClientStreamBridge.convertResponseBody(from: response)
+            let convertedData = try await readAllData(from: resultStream)
+
+            XCTAssertEqual(convertedData, originalData,
+                          "Run \(run) failed (dataSize: \(dataSize))")
+        }
+    }
+
+    func test_convertRequestBody_withNoStream() async throws {
+        let allocator = ByteBufferAllocator()
+        let byteStream = ByteStream.noStream
+
+        let result = try await NIOHTTPClientStreamBridge.convertRequestBody(
+            from: byteStream,
+            allocator: allocator
+        )
+
+        // Convert the body to an async sequence and verify it's empty
+        var totalBytes = 0
+        for try await buffer in result {
+            totalBytes += buffer.readableBytes
+        }
+
+        XCTAssertEqual(totalBytes, 0)
+    }
+
+    func test_convertRequestBody_withData() async throws {
+        let allocator = ByteBufferAllocator()
+        let testData = "Hello, World!".data(using: .utf8)!
+        let byteStream = ByteStream.data(testData)
+
+        let result = try await NIOHTTPClientStreamBridge.convertRequestBody(
+            from: byteStream,
+            allocator: allocator
+        )
+
+        var collectedData = Data()
+        for try await buffer in result {
+            collectedData.append(Data(buffer: buffer))
+        }
+
+        XCTAssertEqual(collectedData, testData)
+    }
+
+    func test_convertRequestBody_withStream() async throws {
+        let allocator = ByteBufferAllocator()
+        let testData = Data(repeating: 42, count: 1000)
```

Comment:
Repeating the same value through the entire test data block opens us up to missing a wide range of errors in transferring the data, no?

(See code for generating random test data above at L26)